scripts/check-monitoring.go

```diff
@@ -0,0 +1,175 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+//go:generate go run check-monitoring.go ../monkit.lock
+
+// +build ignore
+
+package main
+
+import (
+	"fmt"
+	"go/ast"
+	"go/token"
+	"go/types"
+	"io/ioutil"
+	"log"
+	"os"
+	"sort"
+	"strings"
+
+	"golang.org/x/tools/go/packages"
+)
+
+var (
+	monkitPath    = "gopkg.in/spacemonkeygo/monkit.v2"
+	lockFilePerms = os.FileMode(0644)
+)
+
+func main() {
+	pkgs, _ := packages.Load(&packages.Config{
+		Mode: packages.NeedCompiledGoFiles | packages.NeedSyntax | packages.NeedName | packages.NeedTypes | packages.NeedTypesInfo,
+	}, "storj.io/storj/...")
+
+	var lockedFnNames []string
+	for _, pkg := range pkgs {
+		lockedFnNames = append(lockedFnNames, findLockedFnNames(pkg)...)
+	}
+	sortedNames := sortAndUnique(lockedFnNames)
+
+	outputStr := strings.Join(sortedNames, "\n")
+	if len(os.Args) == 2 {
+		file := os.Args[1]
+		if err := ioutil.WriteFile(file, []byte(outputStr+"\n"), lockFilePerms); err != nil {
+			log.Fatalf("error while writing to file \"%s\": %s", file, err)
+		}
+	} else {
+		fmt.Println(outputStr)
+	}
+}
+
+func findLockedFnNames(pkg *packages.Package) []string {
+	var (
+		lockedTasksPos []token.Pos
+		lockedTaskFns  []*ast.FuncDecl
+		lockedFnInfos  []string
+		lockedLines    = make(map[int]struct{})
```

Comment:
This declaration should be in the `for _, file := range pkg.Syntax` scope because otherwise there is a bug in a very specific case.
I reproduced the bug in another branch (https://github.com/ifraixedes/storj-storj/commits/if/review-pr-3114).
See this commit to see the hypothetical but realistic case in https://github.com/ifraixedes/storj-storj/commit/8c656f99fcf9e1f7b1ce8f81d22418bd6b1d2e49

And the commented solution which prevents that to happen: https://github.com/ifraixedes/storj-storj/commit/5404888b097c7a27dc22a6946e5105403077c6d5